### PR TITLE
Support full Cloudinary URLs as the `src` prop

### PR DESCRIPTION
**Pull request by danielolaviobr**
Issue #2 

This PR enables the `src` prop to receive the image cloudiary url. It only retrieves the public id from the url and passes it forward. It passes the version as well, if it is present in the url.

This PR does not solves the transformation case discussed in #9

### CODE REVIEW

**Comment by danielolaviobr** [comment]
@colbyfayock I tried to follow the patterns that I saw on the code, if you have any preferences for the codebase, please let me know so I can make the necessary changes 😄

File: next-cloudinary/src/loaders/cloudinary-loader.js [hunk 1]
```diff
@@ -38,23 +39,25 @@ export function cloudinaryLoader(defaultOptions, cldOptions) {
...
+  const cldImage = cld.image(publicId);
 
   transformationPlugins.forEach(({ plugin }) => {
-    const { options: pluginOptions } = plugin({
-      cldImage,
-      options,
-      cldOptions
-    }) || {};
+    const { options: pluginOptions } =
+      plugin({
```
Comment (colbyfayock) [inline next-cloudinary/src/loaders/cloudinary-loader.js:48]: would you mind removing all of the formatting changes? 😬  guessing it was a linter

File: next-cloudinary/src/lib/cloudinary.js [hunk 2]
```diff
@@ -10,25 +10,43 @@ const cld = new Cloudinary({
...
+ * Or just returns it if it is not recognized as a cloudiary image url.
+ *
+ * @param {string} src The cloudiary url or public id.
+ *
+ * @return {string} The images public id
+ */
+export function getPublicId(src) {
+  if ( src.includes('res.cloudinary.com') ) {
+    const urlParts = src.split('/');
+    const hasVersion = urlParts[urlParts.length - 3].match(/v[0-9]+/);
```
[the remark this answers is not in the document]
Comment (danielolaviobr) [reply next-cloudinary/src/lib/cloudinary.js]: I've reimplemented this method using two different regex patterns to catch when there are transformations, and when there aren't. It satisfies all the examples that you gave and also an extra scenario that has transformations, and an overlay with a nested file as well.

image

[the remark this answers is not in the document]
Comment (colbyfayock) [reply next-cloudinary/src/lib/cloudinary.js]: awesome, thank you!

**Comment by colbyfayock** [comment]
thanks @danielolaviobr looks great! i'll get this merged in. once we have a testing suite added via https://github.com/colbyfayock/next-cloudinary/issues/5 you're welcome to open a new issue and submit a new PR to get tests in for this, other I or someone else will handle it :)

@all-contributors please add @danielolaviobr for code

**Comment by colbyfayock** [comment]
@all-contributors please add @danielolaviobr for code

**Pull request merged**